src/sentry/snuba/tasks.py

```diff
@@ -173,6 +173,9 @@ def delete_subscription_from_snuba(query_subscription_id, **kwargs):
 
     if subscription.subscription_id is not None:
         query_dataset = Dataset(subscription.snuba_query.dataset)
+
+        # Clear the query because we don't need it to delete the subscription
```

Comment:
Is this true? I'm not entirely sure if a query would affect the entity key and result in a mismatch downstream, but if it does then we can look for another way to fix the issue

Reply:
It looks like the entity key is just the `dataset.value`, and doesn't take into account the `query` parameter, so this should be safe
https://github.com/getsentry/sentry/blob/master/src/sentry/search/events/builder/discover.py#L1498C29-L1498C29

Reply:
Actually, touched base with @wmak and this fixes it for _this_ issue, but it doesn't hold true for metrics subscriptions. Instead we opted to add a flag to bypass the issue validation